src/uint/boxed/div_limb.rs

```diff
@@ -0,0 +1,42 @@
+//! Implementation of constant-time division via reciprocal precomputation, as described in
+//! "Improved Division by Invariant Integers" by Niels Möller and Torbjorn Granlund
+//! (DOI: 10.1109/TC.2010.143, <https://gmplib.org/~tege/division-paper.pdf>).
+use crate::{uint::reciprocal::div2by1, BoxedUint, Limb, Reciprocal};
+
+/// Divides `u` by the divisor encoded in the `reciprocal`, and returns
+/// the quotient and the remainder.
+#[inline(always)]
+pub(crate) fn div_rem_limb_with_reciprocal(
+    u: &BoxedUint,
+    reciprocal: &Reciprocal,
+) -> (BoxedUint, Limb) {
+    let (u_shifted, u_hi) = u.shl_limb(reciprocal.shift());
+    let mut r = u_hi.0;
+    let mut q = vec![Limb::ZERO; u.limbs.len()];
+
+    let mut j = u.limbs.len();
+    while j > 0 {
+        j -= 1;
+        let (qj, rj) = div2by1(r, u_shifted.as_limbs()[j].0, reciprocal);
+        q[j] = Limb(qj);
+        r = rj;
+    }
+    (BoxedUint { limbs: q.into() }, Limb(r >> reciprocal.shift()))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{div2by1, Reciprocal};
+    use crate::{Limb, NonZero, Word};
+    #[test]
+    fn div2by1_overflow() {
```

Comment:
Since this test tests a function from `uint::reciprocal`, perhaps it should be located there.

Reply:
This test is actually an exact copy of the same test in `uint::div_limb`. I will delete this one in `boxed/div_limb.rs`.